pkg/auth/oauth/external/github/github.go

```diff
@@ -98,3 +128,65 @@ func (p provider) GetUserIdentity(data *osincli.AccessData) (authapi.UserIdentit
 
 	return identity, true, nil
 }
+
+// getUserOrgs retrieves the organization membership for the user with the given access token.
+func getUserOrgs(token string) (sets.String, error) {
+	// start with the empty set, and the initial org url
+	userOrgs := sets.NewString()
+	orgURL := githubUserOrgURL
+
+	for {
+		// fetch organizations
+		organizations := []githubOrg{}
+		links, err := getJSON(orgURL, token, &organizations)
+		if err != nil {
+			return nil, err
+		}
+		for _, org := range organizations {
+			if len(org.Login) > 0 {
+				userOrgs.Insert(strings.ToLower(org.Login))
+			}
+		}
+
+		// see if we need to page
+		// https://developer.github.com/v3/#link-header
+		nextURL := links["next"]
+		if len(nextURL) == 0 {
+			// no next URL, we're done paging
+			break
+		}
+		if nextURL == orgURL {
+			// identical next URL, break to avoid a loop
+			break
+		}
+		orgURL = nextURL
+	}
+
+	return userOrgs, nil
+}
+
+func getJSON(url string, token string, data interface{}) (map[string]string, error) {
+	req, _ := http.NewRequest("GET", url, nil)
+	req.Header.Set("Authorization", fmt.Sprintf("bearer %s", token))
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("Non-200 response from GitHub API call %s: %d", url, res.StatusCode)
+	}
+
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(body, &data); err != nil {
```

Comment:
`json.Decoder` would allow you to stream and avoid the copy.  Doesn't really matter here, but its a useful thing to know/use.